### ISSUE TITLE
Api gateway authorizer `authorizer_result_ttl_in_seconds` is not set to 0 on first run

### ISSUE THREAD

**Issue by kevinkuszyk** [terraform-providers/terraform-provider-aws#12633]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.24
+ provider.aws v2.55.0
```

### Affected Resource(s)

* `aws_api_gateway_authorizer.api_authorizer`

### Terraform Configuration Files

```
resource "aws_api_gateway_authorizer" "api_authorizer" {
  name                             = "authorizer"
  rest_api_id                      = aws_api_gateway_rest_api.api.id
  authorizer_uri                   = aws_lambda_function.authorizer.invoke_arn
  authorizer_credentials           = aws_iam_role.api_authorizer.arn
  type                             = "REQUEST"
  authorizer_result_ttl_in_seconds = 0
}
```

There is a repo here with the full source:
https://github.com/kevinkuszyk/terraform-aws-apigateway-authorizer-cache

### Expected Behavior

I was expecting the authorizer to be created with the `authorizer_result_ttl_in_seconds` set to zero.

### Actual Behavior

It doesn't seem to be set.  It shows like this in the aws console:

<img width="473" alt="Screenshot 2020-04-02 at 15 56 40" src="https://user-images.githubusercontent.com/2734580/78264056-9d6c7c00-74fa-11ea-8ee7-6a0001df3b85.png">

And a `terraform plan` after applying the config gives this (`300` is the default value):

```
  # aws_api_gateway_authorizer.api_authorizer will be updated in-place
  ~ resource "aws_api_gateway_authorizer" "api_authorizer" {
        authorizer_credentials           = "arn:aws:iam::************:role/api-authorizer"
      ~ authorizer_result_ttl_in_seconds = 300 -> 0
        authorizer_uri                   = "arn:aws:apigateway:eu-west-2:lambda:path/2015-03-31/functions/arn:aws:lambda:eu-west-2:************::function:authorizer/invocations"
        id                               = "7q1ebw"
        identity_source                  = "method.request.header.Authorization"
        name                             = "authorizer"
        provider_arns                    = []
        rest_api_id                      = "gdrd3ai27a"
        type                             = "REQUEST"
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Running `terraform apply` again will make the change, so it appears to be an issue with creating new resources.

### Steps to Reproduce

1. Clone my [example repo](https://github.com/kevinkuszyk/terraform-aws-apigateway-authorizer-cache)
1. Run `terraform apply`
3. Run `terraform plan`

### References

- #705
- #8267
- #9605

### Other notes

I would attempt a fix, but I can't see anything wrong in the code.  If someone can provide some guidance I will do what I can.

- https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_api_gateway_authorizer.go#L71-L75
- https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_api_gateway_authorizer.go#L112-L113

**Comment by kevinkuszyk** [comment]
I've done some testing with the aws cli this morning and i can recreate this issue in the aws console by not setting the `authorizerResultTtlInSeconds` property in my cli command.

/cc @DrFaust92

**Comment by anGie44** [comment]
Fix has been merged and will release with upcoming `v3.0.0` of the AWS Terraform Provider
